src/shogun/neuralnets/RBM.cpp

```diff
@@ -136,6 +137,9 @@ void CRBM::train(CDenseFeatures<float64_t>* features)
 		buffer = SGMatrix<float64_t>(m_num_hidden, m_batch_size);
 
 	int32_t counter = 0;
+
+	auto pb = progress(range(0, max_num_epochs));
+
 	for (int32_t i=0; i<max_num_epochs; i++)
```

Comment:
You could have used the compact version here, instead of the `pb.print_progress/complete` methods :) namely:
`for (auto i : progress(range(0, max_num_epochs)))`

Reply:
Ok, thanks. I will modify the code.